### PR TITLE
Update project to VS 2017

### PR DESCRIPTION
**Pull request by jtaubensee**
## Description
Updating project and build to VS 2017

### CODE REVIEW

**Comment by jtaubensee** [comment]
Im not sure what this part is about: https://ci.appveyor.com/project/jtaubensee/azure-event-hubs-dotnet/build/1.0.311#L105

```
C:\Program Files\dotnet\sdk\1.0.1\Microsoft.Common.CurrentVersion.targets(1964,5): warning MSB3243: No way to resolve conflict between "mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" and "mscorlib". Choosing "mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" arbitrarily. [C:\projects\azure-event-hubs-dotnet\src\Microsoft.Azure.EventHubs\Microsoft.Azure.EventHubs.csproj]
CSC : error CS0006: Metadata file '\mscorlib.dll' could not be found [C:\projects\azure-event-hubs-dotnet\src\Microsoft.Azure.EventHubs\Microsoft.Azure.EventHubs.csproj]
```

**Comment by serkantkaraca** [comment]
:shipit:

**Comment by jtaubensee** [comment]
@onovotny - Do you happen to know what might cause the issue above? The build seems to complete in VS, but when running `dotnet build` the error about an ambiguous mscorlib is thrown. Any help would be appreciated.

**Comment by onovotny** [comment]
I'm not super sure as `dotnet build` should build .NET Desktop libraries without any issues. I would file an issue at https://github.com/dotnet/sdk 

That said, a workaround would be to use msbuild.exe but you shouldn't have to do that in this case.

**Comment by onovotny** [comment]
Oh, sorry, I completely take that back. Because of the UWP target, you have to use MSBuild.exe; dotnet build will not work. Furthermore, you need to use the `MSBuild.SDK.Extras` package and include the require `Include` in your csrpoj as described here: https://github.com/onovotny/MSBuildSdkExtras That's the only way to ensure that UWP is built correctly.

**Comment by jtaubensee** [comment]
@onovotny - Ahh, thank you. I will give that a shot!

File: build/AppveyorBuildFunctions.ps1 [hunk 1]
```diff
@@ -3,10 +3,10 @@ function Build-Solution
     Write-Host "Building Event Hubs projects"
 
     dotnet restore
```
Comment (onovotny) [inline build/AppveyorBuildFunctions.ps1]: that should be `msbuild /t:restore` since there are some project files that `dotnet` can't parse...

**Pull request merged**